### PR TITLE
Use pending_xref for links

### PR DESCRIPTION
**Pull request by pfultz2**
By using `pending_xref` for links, sphinx will resolve links, which works better then trying to duplicate the logic.

### CODE REVIEW

File: recommonmark/parser.py [hunk 1]
```diff
@@ -227,16 +228,30 @@ def inline_entity(inline):
     return entity_node
 
 
+def make_refnode(label, target, has_explicit_title):
```
Comment (ericholscher) [inline recommonmark/parser.py]: Is this logic copied from Sphinx somewhere? Should have a comment specifying where it came from.

Comment (pfultz2) [inline recommonmark/parser.py]: Kinda, it partially came from the `XRefRole` [here](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/roles.py#L127), but just to understand how `pending_xref` worked. The goal is to make it work like the `:any:` role for references except when an url is given. I can add a comment about this.

**Comment by pfultz2** [comment]
So, I added some info to how links are handled in the doc.

**Comment by ericholscher** [comment]
@agjohnson does this interfere with the PR upgrade? Seems to be mostly different, but wanted to make sure before merging it.

**Comment by agjohnson** [comment]
Nope, it does not fit, it would need to be reworked. I can update the my PR for this if you want to merge it.

**Comment by ericholscher** [comment]
Sounds good.

**Comment by brechtm** [comment]
Unfortunately, using _pending_xref_ breaks pure-docutils usage of recommonmark: see #205 and #202.

If you intend recommonmark to be used only in Sphinx projects, can you update the documentation to reflect that?

**Pull request merged**